recipes_source/distributed_comm_debug_mode.rst

```diff
@@ -0,0 +1,202 @@
+Using CommDebugMode
+=====================================================
+
+**Author**: `Anshul Sinha <https://github.com/sinhaanshul>`__
+
+Prerequisites
+
+- Python 3.8 - 3.11
+- PyTorch 2.2 or later
+
+
+What is CommDebugMode and why is it useful
+------------------------------------------
+As the size of models continues to increase, users are seeking to leverage various combinations
+of parallel strategies to scale up distributed training. However, the lack of interoperability
+between existing solutions poses a significant challenge, primarily due to the absence of a
+unified abstraction that can bridge these different parallelism strategies. To address this
+issue, PyTorch has proposed `DistributedTensor(DTensor)
+<https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/examples/comm_mode_features_example.py>`_
+which abstracts away the complexities of tensor communication in distributed training,
+providing a seamless user experience. However, this abstraction creates a lack of transparency
+that can make it challenging for users to identify and resolve issues. To address this challenge,
+``CommDebugMode``, a Python context manager will serve as one of the primary debugging tools for
+DTensors, enabling users to view when and why collective operations are happening when using DTensors,
+effectively addressing this issue.
+
+
+How to use CommDebugMode
+------------------------
+
+Here is how you can use ``CommDebugMode``:
+
+.. code-block:: python
+
+    comm_mode = CommDebugMode()
+        with comm_mode:
+            output = model(inp)
```

Comment:
maybe explain a bit in the comment that `model` here could be a MLP model that applied TP?